packages/node/src/login/oidc/oidcHandlers/ClientCredentialsOidcHandler.ts

```diff
@@ -103,10 +104,24 @@ export default class ClientCredentialsOidcHandler implements IOidcHandler {
     }
 
     if (tokens.id_token === undefined) {
-      throw new Error(
-        `Invalid response from Solid Identity Provider [${
-          oidcLoginOptions.issuer
-        }]: ${JSON.stringify(tokens)} is missing 'id_token'`
+      // In the case where no ID token is provided, the access token is used to
+      // get the authenticated user's WebID. This is only a temporary solution,
+      // as eventually we want to move away from the Identity Provider issuing
+      // Access Tokens, but by then panel work for the bot use case support will
+      // have moved forward.
+      webId = await getWebidFromTokenPayload(
+        tokens.access_token,
+        oidcLoginOptions.issuerConfiguration.jwksUri,
+        oidcLoginOptions.issuer,
+        // When validating the Access Token, the audience should always be 'solid'
+        "solid"
```

Comment:
I added a test in 95b1d746bde583d2ae2dcf845d1a64dcd36b0b20, although since token validation normally requires a network lookup, it is mocked, so I don't know how much value the test adds.